manifests/config.pp

```diff
@@ -102,7 +102,7 @@
   # Just convert to yaml
   file {"${config_path}/${config_file}":
     ensure  => $ensure,
-    content => inline_template('<%= @cobbler_config.to_yaml %>'),
```

Comment:
In addition to random order in hashes, the current issue is also in interfaces configuration comparison. On every puppet run you get something like this:
../Cobbler_system[host1.tesdomain.com]/interfaces: p2p2: interface:  -> p2p2, p2p1: interface:  -> p2p1, em2: interface:  -> em2, em1: interface:  -> em1, br0: interface:  -> br0, bond0: interface:  -> bond0

The issue is buried in way how ruby Ruby hashes random ordering... Need a fix.